api/osb/logic.go

```diff
@@ -205,15 +205,39 @@ func clientConfigForBroker(broker *types.Broker) *osbc.ClientConfiguration {
 func (b *BusinessLogic) osbClient(request *http.Request) (osbc.Client, error) {
 	vars := mux.Vars(request)
 	brokerID, ok := vars[BrokerIDPathParam]
-	logrus.Debugf("Obtained path parameter [brokerID = %s] from mux vars", brokerID)
 	if !ok {
-		return nil, fmt.Errorf("error creating OSB client: brokerID path parameter not found")
+		logrus.Debugf("error creating OSB client: brokerID path parameter not found")
+		return nil, createHTTPErrorResponse("Invalid broker id path parameter", "BadRequest", http.StatusBadRequest)
 	}
+	logrus.Debugf("Obtained path parameter [brokerID = %s] from mux vars", brokerID)
+
 	serviceBroker, err := b.brokerStorage.Get(brokerID)
-	if err != nil {
-		return nil, fmt.Errorf("error obtaining serviceBroker with id %s from storage: %s", brokerID, err)
+	if err == storage.ErrNotFound {
+		logrus.Debugf("service broker with id %s not found", brokerID)
+		return nil, createHTTPErrorResponse(fmt.Sprintf("Could not find broker with id: %s", brokerID), "NotFound", http.StatusNotFound)
+	} else if err != nil {
+		logrus.Errorf("error obtaining serviceBroker with id %s from storage: %s", brokerID, err)
+		return nil, fmt.Errorf("Internal Server Error")
 	}
 	config := clientConfigForBroker(serviceBroker)
 	logrus.Debug("Building OSB client for serviceBroker with name: ", config.Name, " accessible at: ", config.URL)
 	return b.createFunc(config)
 }
+
+func toHTTPError(err error) error {
+	if err == nil {
```

Comment:
While it is ok for the `osbc.IsHttpError` to be called with nil we still need to check for it as the logic of the code is not to do anything in the case of nil. Another approach would be to call `osbc.IsHttpError` and check if the returned error is nil. But this will be roughly the same.